backendCore/common/ai_drive.py

```diff
@@ -0,0 +1,272 @@
+    """
+    try:
+        if dataset not in DEFAULT_DATASETS:
+            raise Exception(
+                f"The {dataset} file does not currently exist in our inventory. Please submit a request to the contributors of the repository"
+            )
+        else:
+            raw_data = eval(
+                DEFAULT_DATASETS[dataset]
+            )  # get raw data from sklearn.datasets
+            target_names = []
+            try:
+                target_names = list(raw_data["target_names"])
+            except:
+                pass
+            default_dataset = pd.DataFrame(
+                data=np.c_[raw_data["data"], raw_data["target"]],
+                columns=raw_data["feature_names"] + ["target"],
+            )
+            # remove any empty lines
+            default_dataset.dropna(how="all", inplace=True)
+            if features and target:
+                y = default_dataset[target]
+                X = default_dataset[features]
+            else:
+                y = default_dataset["target"]
+                X = default_dataset.drop("target", axis=1)
+            print(default_dataset.head())
+            return X, y, target_names
+    except Exception:
+        raise Exception(f"Unable to load the {dataset} file into Pandas DataFrame")
+
+
+def get_tensors(X_train, X_test, y_train, y_test):
+    """
+    Helper function to convert X_train, X_test, y_train, y_test
+    into tensors for dataloader
+    Args:
+        X_train (pd.DataFrame): X_train (train set)
+        X_test (pd.DataFrame): X_test (test set)
+        y_train (pd.Series): label/value of target for each row of X_train
+        y_test (pd.Series): label/value of target for each row of X_test
+
+    Return:
+        X_train, X_test, y_train, y_test in the form of tensor
+    """
+    X_train_tensor = Variable(torch.Tensor(X_train.to_numpy()))
+    y_train_tensor = Variable(torch.Tensor(y_train.to_numpy()))
+    X_train_tensor = torch.reshape(
+        X_train_tensor, (X_train_tensor.size()[0], 1, X_train_tensor.size()[1])
+    )
+    y_train_tensor = torch.reshape(y_train_tensor, (y_train_tensor.size()[0], 1))
+
+    X_test_tensor = Variable(torch.Tensor(X_test.to_numpy()))
+    y_test_tensor = Variable(torch.Tensor(y_test.to_numpy()))
+    X_test_tensor = torch.reshape(
+        X_test_tensor, (X_test_tensor.size()[0], 1, X_test_tensor.size()[1])
+    )
+    y_test_tensor = torch.reshape(y_test_tensor, (y_test_tensor.size()[0], 1))
+
+    X_train_tensor.requires_grad_(True)
+    X_test_tensor.requires_grad_(True)
+    return X_train_tensor, y_train_tensor, X_test_tensor, y_test_tensor
+
+
+def get_optimizer(model, optimizer_name, learning_rate):
+    """
+    Given an optimizer name, instantiate the object
+
+    Args:
+        model (nn.Module): pytorch model to train
+        optimizer_name (str): name of optimizer
+        learning_rate (float): learning rate
+    """
+    if optimizer_name.upper() == "SGD":
```

Comment:
potential code smell here. If we add another type of optimizer, we end up adding an extra if/elif statement. In future features, that type of behavior might not be extensible. We don't have to worry about that rn, but pointing it out for the benefit of the team